src/stateFromElement.js

```diff
@@ -251,8 +264,12 @@ class BlockGenerator {
       styleStack: [NO_STYLE],
       entityStack: [NO_ENTITY],
       depth: hasDepth ? this.depth : 0,
+      data: {},
     };
     if (allowRender) {
+      if (this.options && this.options.hasOwnProperty('filterBlockData') && this.options.filterBlockData) {
+        block.data = fromJS(this.options.filterBlockData({}, element));
```

Comment:
Oh, also we don't need `fromJS` because block.data in this case should be a plain object.